shotover-proxy/src/transforms/chain.rs

```diff
@@ -291,6 +290,34 @@ impl TransformChain {
         histogram!("shotover_chain_latency", start.elapsed(),  "chain" => self.name.clone(), "client_details" => client_details);
         result
     }
+
+    pub async fn process_request_rev(&mut self, mut wrapper: Wrapper<'_>) -> ChainResponse {
+        let start = Instant::now();
+        wrapper.reset_reverse(&mut self.chain);
```

Comment:
This call actually reverses the order of self.chain.
That sounds wrong, wont that affect the order of non pushed messages?

Reply:
I think this would be the best fix:
![image](https://user-images.githubusercontent.com/5120858/176100981-353586a0-b915-430e-9179-0c75e0d0f708.png)

Reply:
Yeah, you are correct I believe. That code was written when the chain was being cloned before calling that method so the code was leftover.